akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -181,7 +181,12 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
   def completeConnect(registration: ChannelRegistration, commander: ActorRef,
                       options: immutable.Traversable[SocketOption]): Unit = {
     // Turn off Nagle's algorithm by default
-    channel.socket.setTcpNoDelay(true)
+    try channel.socket.setTcpNoDelay(true) catch {
+      case e: SocketException ⇒
+        // as reported in #16653 some versions of netcat (`nc -z`) doesn't allow setTcpNoDelay
```

Comment:
Hm. Isn't the problem maybe that nc immediately closes the connection after a handshake? Can't this option set pre-connect?